packages/salesforcedx-vscode-core/src/commands/util/sfdxCommandlet.ts

```diff
@@ -125,6 +168,77 @@ export abstract class SfdxCommandletExecutor<T>
     return;
   }
 
+  public outputResultPull(parser: ForcePullResultParser) {
+    const table = new Table();
+    const titleType = 'pull';
+
+    const successes: any = parser.getSuccesses();
+    const errors = parser.getErrors();
+    const pulledSource = successes ? successes.result.pulledSource : undefined;
+    if (pulledSource) {
+      const rows = pulledSource || (errors && errors.result);
+      const tableTitle = nls.localize(`table_title_${titleType}ed_source`);
+      const outputTable = this.getOutputTable(table, rows, tableTitle);
+
+      channelService.appendLine(outputTable);
+      if (pulledSource && pulledSource.length === 0) {
```

Comment:
In this flow does it show an empty table and then the empty message?

Reply:
Nope, here it looks like this:
<img width="422" alt="2023-05-19_17-34-28" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/46458081/193ea4d0-090f-48a1-9dfe-82d166ca49b8">